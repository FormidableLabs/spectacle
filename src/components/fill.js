import PropTypes from 'prop-types';
import styled from 'react-emotion';

export const Fill = styled.div`
  flex: 1;
`;

Fill.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};
