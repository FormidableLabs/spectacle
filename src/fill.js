import React, { Component, PropTypes } from "react";
import Radium from "radium";

@Radium
class Fill extends Component {
  render() {
    const styles = {
      flex: 1
    };
    return (
      <div style={[styles, this.props.style]}>
        {this.props.children}
      </div>
    );
  }
}

Fill.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object
};

export default Fill;
