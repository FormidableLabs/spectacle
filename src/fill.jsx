import React from "react/addons";
import Radium from "radium";

@Radium
class Fill extends React.Component {
  render() {
    const styles = {
      flex: 1
    };
    return (
      <div style={[styles]}>
        {this.props.children}
      </div>
    );
  }
}

Fill.propTypes = {
  children: React.PropTypes.node
};

export default Fill;
