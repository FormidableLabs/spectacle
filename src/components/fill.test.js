import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { Fill } from './fill';

describe('<Fill />', () => {
  test('should render with style `flex: 1`', () => {
    const wrapper = mount(<Fill>Spectacle</Fill>);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
