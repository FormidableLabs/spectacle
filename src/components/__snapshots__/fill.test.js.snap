// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fill /> should render with style \`flex: 1\` 1`] = `
<Styled(div)>
  <div
    className="css-1rr4qq7"
  >
    Spectacle
  </div>
</Styled(div)>
`;
